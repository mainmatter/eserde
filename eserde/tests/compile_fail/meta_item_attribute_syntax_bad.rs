#[derive(eserde::Deserialize)]
struct Foo {
    #[serde(default - this parses but is not Meta Item Attribute Syntax, serde errors "expected `,`")]
    field: u32,
}

fn main() {}
