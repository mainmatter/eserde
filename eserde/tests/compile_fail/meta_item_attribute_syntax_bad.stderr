error: expected `,`
 --> tests/compile_fail/meta_item_attribute_syntax_bad.rs:3:21
  |
3 |     #[serde(default - this parses but is not Meta Item Attribute Syntax, serde errors "expected `,`")]
  |                     ^

error: duplicate serde attribute `default`
 --> tests/compile_fail/meta_item_attribute_syntax_bad.rs:3:5
  |
3 |     #[serde(default - this parses but is not Meta Item Attribute Syntax, serde errors "expected `,`")]
  |     ^

error[E0277]: the trait bound `__ImplEDeserializeForFoo: Deserialize<'_>` is not satisfied
 --> tests/compile_fail/meta_item_attribute_syntax_bad.rs:2:8
  |
2 | struct Foo {
  |        ^^^ the trait `Deserialize<'_>` is not implemented for `__ImplEDeserializeForFoo`
  |
  = note: for local types consider adding `#[derive(serde::Deserialize)]` to your `__ImplEDeserializeForFoo` type
  = note: for types from other crates check whether the crate offers a `serde` feature flag
  = help: the following other types implement trait `Deserialize<'de>`:
            &'a [u8]
            &'a std::path::Path
            &'a str
            ()
            (T,)
            (T0, T1)
            (T0, T1, T2)
            (T0, T1, T2, T3)
          and $N others

error[E0599]: no function or associated item named `deserialize` found for struct `__ImplDeserializeForFoo` in the current scope
 --> tests/compile_fail/meta_item_attribute_syntax_bad.rs:1:10
  |
1 | #[derive(eserde::Deserialize)]
  |          ^^^^^^^^^^^^^^^^^^^ function or associated item not found in `__ImplDeserializeForFoo`
2 | struct Foo {
  | ---------- function or associated item `deserialize` not found for this struct
  |
  = help: items from traits can only be used if the trait is implemented and in scope
  = note: the following traits define an item `deserialize`, perhaps you need to implement one of them:
          candidate #1: `Deserialize`
          candidate #2: `DeserializeSeed`
  = note: this error originates in the derive macro `eserde::Deserialize` (in Nightly builds, run with -Z macro-backtrace for more info)
